src/smartbanner.js

```diff
@@ -157,7 +190,17 @@ export default class SmartBanner {
     return Detector.userAgentMatchesRegex(this.options.includeUserAgentRegex);
   }
 
-  publish() {
+  publish(ignoreMetas) {
```

Comment:
Publish method as the main method should remain clean without any parameters. It's important to keep API clean. `Detector` class should be responsible for any detection, this method here just deals with publishing, not detection.